java/arcs/core/data/RawEntity.kt

```diff
@@ -24,22 +24,32 @@ data class RawEntity(
      * Collection ([Set]) fields and the set of [ReferenceId]s referencing the values in those
      * collections.
      */
-    val collections: Map<FieldName, Set<Referencable>> = emptyMap(),
-    /** Indication of the timestamp when this entity expires. */
-    @Suppress("GoodTime") // use Instant
-    val expirationTimestamp: Long = NO_EXPIRATION
+    val collections: Map<FieldName, Set<Referencable>> = emptyMap()
 ) : Referencable {
     override fun tryDereference(): Referencable {
-        return RawEntity(
+        val entity = RawEntity(
             id = id,
-            expirationTimestamp = expirationTimestamp,
             singletons = singletons.mapValues { it.value?.tryDereference() },
             collections = collections.mapValues {
                 it.value.map { item -> item.tryDereference() }.toSet()
             }
         )
+        entity.expirationTimestamp = expirationTimestamp
+        return entity
     }
 
+    /** Entity expiration time (in millis). */
+    @Suppress("GoodTime") // use Instant
+    override var expirationTimestamp: Long = NO_EXPIRATION
+        get() = field
```

Comment:
No need for this getter if it's just returning the `field` value.

Reply:
removed